File: skyspace222_ui/src/services/RoutePlanService.js
import http from "../http-common"; 

class RoutePlanService {
  getAllRoutePlans(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/routePlan/routePlans`, searchDTO);
  }

  get(routePlanId) {
    return this.getRequest(`/routePlan/${routePlanId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/routePlan?field=${matchData}`, null);
  }

  addRoutePlan(data) {
    return http.post("/routePlan/addRoutePlan", data);
  }

  update(data) {
  	return http.post("/routePlan/updateRoutePlan", data);
  }
  
  uploadImage(data,routePlanId) {
  	return http.postForm("/routePlan/uploadImage/"+routePlanId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RoutePlanService();

File: src/main/java/com/skyspace222/service/RoutePlanService.java
package com.skyspace222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace222.domain.RoutePlan;
import com.skyspace222.dto.RoutePlanDTO;
import com.skyspace222.dto.RoutePlanSearchDTO;
import com.skyspace222.dto.RoutePlanPageDTO;
import com.skyspace222.dto.RoutePlanConvertCriteriaDTO;
import com.skyspace222.service.GenericService;
import com.skyspace222.dto.common.RequestDTO;
import com.skyspace222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RoutePlanService extends GenericService<RoutePlan, Integer> {

	List<RoutePlan> findAll();

	ResultDTO addRoutePlan(RoutePlanDTO routePlanDTO, RequestDTO requestDTO);

	ResultDTO updateRoutePlan(RoutePlanDTO routePlanDTO, RequestDTO requestDTO);

    Page<RoutePlan> getAllRoutePlans(Pageable pageable);

    Page<RoutePlan> getAllRoutePlans(Specification<RoutePlan> spec, Pageable pageable);

	ResponseEntity<RoutePlanPageDTO> getRoutePlans(RoutePlanSearchDTO routePlanSearchDTO);
	
	List<RoutePlanDTO> convertRoutePlansToRoutePlanDTOs(List<RoutePlan> routePlans, RoutePlanConvertCriteriaDTO convertCriteria);

	RoutePlanDTO getRoutePlanDTOById(Integer routePlanId);







}





